### DISCUSSION TITLE
ESP8266 Wemos Nodemcu v3 ESP-01 + Max485 converter with Flow Control Pins + Must Pro PV18 5248 inverter

### DISCUSSION THREAD

**Discussion by Rogero-** [vladyspavlov/esphome-must-inverter#9] [Q&A] [answered]
hello, i'm trying from few days to make this project work with my Must PV-18 5248, i am using a Nodemcu v3 with ESP01

![wifi-module-esp8266-wemos-nodemcu-v3](https://github.com/user-attachments/assets/1d22679a-f6ed-4f15-b587-6454442c2de1)

I am using the RX/TX pads to connect it with the MAX485 board and i know i can't use LOG because of the shared RX pin with the UART TTL onboard converter on the Nodemcu. 

in your original post you used a different RS485 conversion board that only have RX/TX and VCC/GND pins to connect with the ESP board, i use this MAX485 board that uses 5v input and not 3.3v

![MAX485 5V CHIP-RS485-Module-Pinout-](https://github.com/user-attachments/assets/b1bf7992-6d06-4ce0-8f41-b9e54f34dc63)

it have those 2 additional Pins DE/RE that must be pulled high to enable the driver, with your yaml configuration which pad or pads on the ESP should DE/RE be connected to  ( and should both pins be connected together to the same pad on the ESP ? )

i tried connecting both pins together to D5 pad on ESP which seems to be pulled HIGH by default (5v) but still i can't get a proper output from the inverter.

Please any help or directions will be appreciated, i have to make it work with Home Assistant, thx :)

**Comment by vladyspavlov** [comment, marked as answer]
You should use 3.3V instead of the 5V, it affects the TTL voltage. Easy way is to take it from the ESP.

Here's some info about the RE/DE
https://electronics.stackexchange.com/questions/153500/correctly-using-re-and-de-with-rs485

Look at the https://esphome.io/components/modbus.html#configuration-variables
Try to add `flow_control_pin: GPIO0` to the modbus controller config and connect RE/DE to the D3 (which is GPIO0?).

**Comment by Rogero-** [comment]
Thank you man for your help about the "flow_control_pin" this is what was missing to make my MAX485 board work properly, i added the Flow Control code to the yaml config under "modbus:" and "not modbus controller:" like this:

![image](https://github.com/user-attachments/assets/850da073-5c43-4bf1-8d12-cecbd8dd2d12)

this is the updated yaml file that works with this MAX485 board (with Flow control pins):
[esp01-pv18-max485-with-Flow-Control.yaml.txt](https://github.com/user-attachments/files/18166670/esp01-pv18-max485-with-Flow-Control.yaml.txt)

and connected RE/DE to D3/GPIO0 on ESP and i powered the RS485 board with 3.3v from the ESP (although this MAX485 ic works on 5v but it seems working fine on 3.3v also and better to avoid problems with the TTL voltage as you instructed)  All is working now, i'm soooo happy finally it works, i will cleanup my wiring now since i know the correct schema, and i will upload the proper connection diagram for others using the same hardware (RS485 MAX485 board with Flow control pins + ESP01/ESP8266)

here is the working connections diagram:

![ESP8266 ESP01 and MAX485 board with Flow Control Pins](https://github.com/user-attachments/assets/bb8cfdd6-6f80-493f-8cfd-a4a896973477)

and some screenshots :)

![ESPHome Device](https://github.com/user-attachments/assets/6547b2b8-35cf-4c4c-b976-f2dcc08dc171)
![Must PV18 5248 Connected](https://github.com/user-attachments/assets/ea2def39-a7b5-4655-a4d0-416df518c75e)
![20241217_172134](https://github.com/user-attachments/assets/2b4bbc2a-cc3d-4280-8724-43ccb405ec01)
![20241217_172212](https://github.com/user-attachments/assets/9946f472-87d9-481f-b41c-fbaf66aaeee9)
![20241217_172517](https://github.com/user-attachments/assets/bbaa1a41-b474-4129-9bba-0f7e4385ce6e)
![20241217_172533](https://github.com/user-attachments/assets/a2b480d1-4e33-4f58-b800-d2b5bc94bef8)
![20241217_172814](https://github.com/user-attachments/assets/8adf9d6c-e447-49c4-a0fa-d617ed15cb7a)
![20241217_172827](https://github.com/user-attachments/assets/11844dbc-fe3d-4833-a4bb-0fb35999867b)
